### PR TITLE
CMSystemDiscovery - Adding Parameter Validation

### LINKED ISSUE #73: CMSystemDiscovery parameter handling
CMSystemDiscovery parameter handling
#### Details of the scenario you tried and the problem that is occurring

The TimeSinceLastLogonDays Parameter: underlying commandlet only accepts 14-720. Suggest a validaterange.
The TimeSinceLastPasswordUpdateDays: the underlying commandlet only accepts 30-720. Suggest a validaterange.

The schedule count needs warnings and handling when the maximum allowed interval is exceeded by input, similar to how we are handling the Heartbeat discovery.

### PR DESCRIPTION
**Pull request by NEllis280**
#### Pull Request (PR) description
This PR adds warnings to the CMSystemDiscovery resource warning when the maximum schedule interval has been exceeded or the minimum not met (for minutes) and adds handling for test/set when the specified interval is greater/less than the maximum/minimum allowed.

Also adds a validaterange on the TimeSinceLastPasswordUpdateDays and TimeSinceLastLogonDays parameters.

#### This Pull Request (PR) fixes the following issues
- Fixes #73 

#### Task list
- [x] Added an entry to the change log under the Unreleased section of the
      file CHANGELOG.md. Entry should say what was changed and how that
      affects users (if applicable), and reference the issue being resolved
      (if applicable).
- [ ] Resource documentation added/updated in README.md.
- [ ] Resource parameter descriptions added/updated in README.md, schema.mof
      and comment-based help.
- [ ] Comment-based help added/updated.
- [x] Localization strings added/updated in all localization files as appropriate.
- [ ] Examples appropriately added/updated.
- [x] Unit tests added/updated. See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [ ] Integration tests added/updated (where possible). See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [x] New/changed code adheres to [DSC Community Style Guidelines](https://dsccommunity.org/styleguidelines).

---
This change is [Reviewable](https://reviewable.io/reviews/dsccommunity/configmgrcbdsc/80)

### CODE REVIEW

File: source/DSCResources/DSC_CMSystemDiscovery/DSC_CMSystemDiscovery.psm1 [hunk 1]
```diff
@@ -265,15 +267,43 @@ function Set-TargetResource
...
+                {
+                    Write-Warning -Message ($script:localizedData.MaxIntervalDays -f $ScheduleCount)
+                    $scheduleCheck = 31
+                }
+                elseif ($ScheduleInterval -eq 'Hours' -and $ScheduleCount -ge 24)
+                {
+                    Write-Warning -Message ($script:localizedData.MaxIntervalHours -f $ScheduleCount)
+                    $scheduleCheck = 23
+                }
+                elseif (($ScheduleInterval -eq 'Minutes' -and $ScheduleCount -ge 60) -or ($ScheduleInterval -eq 'Minutes' -and $ScheduleCount -le 4))
```
Comment (jeffotterpohl) [inline source/DSCResources/DSC_CMSystemDiscovery/DSC_CMSystemDiscovery.psm1:280]: could shrink this up a little bit
`elseif (($ScheduleInterval -eq 'Minutes') -and ($ScheduleCount -ge 60 -or $ScheduleCount -le 4))`

**Comment by jeffotterpohl** [comment]
I know it would be more work, any thought of going ahead and also converting the ADContainers\Include\Exclude over to the helper function Compare-MultipleCompares (example can be found in DSC_CMSoftwareDistributionComponent.  This would reduce the amount of code and bring it into more of a standardized way of comparing in this resource.

File: source/DSCResources/DSC_CMSystemDiscovery/DSC_CMSystemDiscovery.psm1 (hunk 1 above)
Comment (NEllis280) [reply source/DSCResources/DSC_CMSystemDiscovery/DSC_CMSystemDiscovery.psm1]: Done.

**Comment by NEllis280** [comment]
Need to update PrimaryInstall.ps1 example with changes

**Comment by jeffotterpohl** [comment]
Code looks good just need to finish testing and will approve.

**Comment by jeffotterpohl** [comment]
Looks good and works accordingly

**Review by jeffotterpohl — APPROVED** [review]
Good to go

**Pull request merged**